replica/database.cc

```diff
@@ -1399,6 +1399,10 @@ bool database::can_apply_per_partition_rate_limit(const schema& s, db::operation
     return replica::can_apply_per_partition_rate_limit(s, _dbcfg, op_type);
 }
 
+bool database::is_internal_query() const {
+    return classify_request(_dbcfg) != request_class::user;
+}
```

Comment:
Why make the distinction?

If a system query is paged (as it should be but isn't) it should tolerate empty pages.

If it's unpaged, the query tombstone page limit should be unlimited.

Reply:
I wanted to follow the behavior of max result size here: when reached cut the page for paged queries and abort on unpaged ones.

Reply:
Perhaps the internal query client can set a max limit, and avoid this if maze.

Reply:
This is much more work, requires accurately tracking down all places initiating internal queries. It is much better to delegate the decision to a single entity (storage proxy), who makes the decision based on the query class. This is the same method we used for max result size.